### ISSUE TITLE
round of new 2.12 failures

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#704]
consistently failing in [recent runs](https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/): fs2, ​play-core, ​refined, ​ammonite

and there's also a new circe failure (compilation fails) in [run 2714](https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2714/consoleFull)

**Comment by SethTisue** [comment]
fs2 is a compilation failure: "value shift is not a member of object cats.effect.Async" — presumably solvable by freezing fs2 and/or cats-effect at the last green versions

play-core is a compilation failure: "framework/src/play-ahc-ws/src/main/scala/play/api/libs/ws/ahc/AhcWSResponse.scala:19: class AhcWSResponse needs to be abstract, since method uri in trait StandaloneWSResponse of type => java.net.URI is not defined" — probably fixable by freezing play-core and/or play-ws for now

refined is "Missing bintray credentials", presumably fixable with the usual workaround. (not clear to me why it started all of a sudden, there have been some recent changes to the build though, the exact causation chain isn't clear but probably not worth investigating)

ammonite is "amm/util/src/main/scala/ammonite/util/Imports.scala:124: value readwriter is not a member of object upickle.default". probably it's because of the recent big upickle overhaul and ammonite is still using the old one? if so, fixable by freezing upickle

**Comment by SethTisue** [comment]
circe failure is "/modules/literal/src/test/scala/io/circe/literal/interpolator/JsonInterpolatorSuite.scala:18:9: Invalid JSON in interpolated string" — seems rather mysterious since the most recent circe commit is Mar 19 and we haven't bumped the Scala version here in this repo recently

it failed again in run 2715 so not transient

**Comment by SethTisue** [comment]
last totally green run was https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2672/ (Mar 17)

> probably it's because of the recent big upickle overhaul and ammonite is still using the old one

wrong guess — actually ammonite has upgraded upickle in recent weeks (multiple times). so we either need to freeze ammonite for now, or upgrade our upickle. in general we don't like to upgrade low level dependencies like upickle that have made breaking changes, because doing so is difficult/impossible until the community has first upgraded for their own reasons, so I think we should probably just freeze ammonite. /cc @lihaoyi 

as for the monix/fs2/cats-effect situation, it looks both monix and fs2 have moved from cats-effect 0.9 (which we're currently frozen on) to 0.10, so it should work to move cats-effect to 0.10 and unfreeze monix and fs2. (but I'm tempted to just freeze all three, since they are co-evolving rapidly.)

**Comment by SethTisue** [comment]
although I haven't looked _too_ closely at it, the culprit commit in play-ws is probably playframework/play-ws@e84d4cd891880e289f7e1fb158093f874c6ece8b, so freezing play-ws at ~~playframework/play-ws@c521ad1b975e9a3287179436897a0e08c47a3e56~~ some earlier commit until playframework is upgraded should do the trick

**Comment by SethTisue** [comment]
testing these theories at #706

**Comment by SethTisue** [comment]
as for circe,

`[info] Project circe depends on: catalysts, cats, cloc-plugin, discipline, jawn-0-11, json4s, kind-projector, machinist, macro-compat, macro-paradise, scala, scala-parser-combinators, scalacheck, scalatest, scalaz, scodec-bits, shapeless, simulacrum, specs2, spray-json, wartremover`

do I need to go groveling through this list and try and figure out what change in a dependency might have caused this...? I guess. note that circe was green in [run 2712](https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2712/) at circe/circe@81ac123d5058afa0da9fb6dc44df1287d69c463e

**Comment by SethTisue** [comment]
the circe failure is the `literal` subproject, and

```
sbt:circe> show literal/libraryDependencies
[info] * org.scala-lang:scala-library:2.12.5
[info] * org.scala-lang:scala-compiler:2.12.5:provided
[info] * org.scala-lang:scala-reflect:2.12.5:provided
[info] * org.typelevel:macro-compat:1.1.1
[info] * org.scalamacros:paradise:2.1.1:plugin->default(compile)
[info] * com.chuusai:shapeless:2.3.3:test
```

this isn't much. there are very recent commits in shapeless, so maybe try a slightly older commit like milessabin/shapeless@546ec68fc6fcc3b7c6c33caf8ac6ce2a4d18cbb1

**Comment by SethTisue** [comment]
#706 made some progress. new status is `failed: monocle, circe, lagom`

* circe: need a new theory
* monocle: is hitting a new deprecation in scalaz; we just need to disable `-Xfatal-warnings`
* lagom: persistence-jdbc-core subproject has missing dependency com.github.dnvriend#akka-persistence-jdbc. unrelated to changes in this PR, I think. probably we should just disable the subproject? (we don't need every integration)

EDIT: on second thought, we ought to at least try adding akka-persistence-jdbc, maybe it will be easy. Lagom is core stuff for Lightbend.

**Comment by SethTisue** [comment]
after #706 and #707, just circe still fails

**Comment by SethTisue** [comment]
the circe issue isn't reproducible outside dbuild with `++2.12.6-bin-84112e8!` and `literal / test` (and changing `"paradise" % paradiseVersion cross CrossVersion.patch` to `"paradise_2.12.5" % paradiseVersion` everywhere in build.sbt)

**Comment by SethTisue** [comment]
here's everything on the test classpath for `literal`:

```
com/chuusai/shapeless_2.12/2.3.3/shapeless_2.12-2.3.3.jar
org/scala-lang/modules/scala-xml_2.12/1.0.6/scala-xml_2.12-1.0.6.jar
org/scala-lang/scala-compiler/2.12.6-bin-84112e8/scala-compiler-2.12.6-bin-84112e8.jar
org/scala-lang/scala-library/2.12.6-bin-84112e8/scala-library-2.12.6-bin-84112e8.jar
org/scala-lang/scala-reflect/2.12.6-bin-84112e8/scala-reflect-2.12.6-bin-84112e8.jar
org/scala-sbt/test-interface/1.0/test-interface-1.0.jar
org/scalacheck/scalacheck_2.12/1.13.5/scalacheck_2.12-1.13.5.jar
org/scalactic/scalactic_2.12/3.0.5-M1/scalactic_2.12-3.0.5-M1.jar
org/scalatest/scalatest_2.12/3.0.5-M1/scalatest_2.12-3.0.5-M1.jar
org/spire-math/jawn-parser_2.12/0.11.1/jawn-parser_2.12-0.11.1.jar
org/typelevel/catalysts-macros_2.12/0.0.5/catalysts-macros_2.12-0.0.5.jar
org/typelevel/catalysts-platform_2.12/0.0.5/catalysts-platform_2.12-0.0.5.jar
org/typelevel/cats-core_2.12/1.1.0/cats-core_2.12-1.1.0.jar
org/typelevel/cats-kernel-laws_2.12/1.1.0/cats-kernel-laws_2.12-1.1.0.jar
org/typelevel/cats-kernel_2.12/1.1.0/cats-kernel_2.12-1.1.0.jar
org/typelevel/cats-laws_2.12/1.1.0/cats-laws_2.12-1.1.0.jar
org/typelevel/cats-macros_2.12/1.1.0/cats-macros_2.12-1.1.0.jar
org/typelevel/discipline_2.12/0.9.0/discipline_2.12-0.9.0.jar
org/typelevel/machinist_2.12/0.6.2/machinist_2.12-0.6.2.jar
org/typelevel/macro-compat_2.12/1.1.1/macro-compat_2.12-1.1.1.jar
```

**Comment by SethTisue** [comment]
[fixed circe](26e9b5b67250eb14ed2d0c92ed6b263d20398dc8) by freezing jawn, fyi @travisbrown that circe breaks if jawn dependency is upgraded to 0.12.1

**Comment by SethTisue** [comment]
whole build is green again after a few more fixes:

* 08fe468 - fork scapegoat
* b18f32c - freeze http4s
* 5ca98c5 - fix Slick by excluding new subprojects
